### DISCUSSION TITLE
Login with multiple providers

### DISCUSSION THREAD

**Discussion by tanshunyuan** [supabase/supabase#1168] [Questions]
Hi there, is it possible to login with multiple providers such as github and gitlab. I would like to mimic's [vercel](https://vercel.com/login) login system. 
Currently when I login with github, the app_metadata displays `github` as my provider.  
![image](https://user-images.githubusercontent.com/29978940/114970413-fe1fea00-9eac-11eb-8089-c864e3d49a5d.png)
However, when I switch to login with `gitlab` the app_metadata still displays `github` as my provider. Is there a way to update it so that the user's provider change everytime the user use a different login provider?

**Comment by kiwicopple** [comment]
Hi @tanshunyuan - at the moment this isn't possible, but it's somethign we are considering (on github.com/supabase/gotrue)

**Reply by tanshunyuan** [reply]
Ok. I have a unrelated question, hoping that probably someone can shed some light on it. With not being able to mimic vercel's login system at the moment, is there a work around it?

Currently, I am developing a website that allow users to retrieve repository from both github or gitlab. However, I am stuck with the authentication flow as it seems like there is no way to mimic's vercel login system currently. 
An alternative I was thinking of is letting users sign in with email, then allow them to re-authenticate/install github or gitlab to obtain access tokens to retrieve data from github or gitlab.

**Reply by kiwicopple** [reply]
You'll need to add some `repository` scopes if you want access to the users' repositories like Vercel: 
https://supabase.io/docs/reference/javascript/auth-signup#sign-up-with-scopes

The other workaround is just to use the GitLab/GitHub APIs directly (so you would need to set up your own OAuth flow

**Reply by tanshunyuan** [reply]
So by passing in scopes I can access users's repository, but there isn't a way to track which provider they are currently using right? Since supabase only registers one provider. Eg switching between github or gitlab accounts to import different repositories.

For the workaround you mention, by implementing my OAuth flow does it mean that I cannot use supabase authentication? And I will have to manually generate and issue jwt tokens and refresh tokens?

Thank you so much for taking your time to answer my question!

**Reply by kiwicopple** [reply]
> For the workaround

You could still use Supabase for the login/Auth, but for managing repository access you would use the APIs provided by GitHub and GitLab - you should dig into their docs for this because they will do a much better job of explaining.

**Reply by tanshunyuan** [reply]
Ok so for now its just that when I switch between github or gitlab, there will not be an update to the providers indicating that the users is logged in with github or gitlab right?
